scripts/dor-soak.py

```
@@ -0,0 +1,284 @@
+
+def read_events(path, since_index):
+    """Journal events after the given line index; tolerates partial lines."""
+    if path is None or not path.exists():
+        return [], since_index
+    lines = path.read_text(errors="replace").splitlines()
+    events = []
+    for line in lines[since_index:]:
+        try:
+            events.append(json.loads(line))
+        except json.JSONDecodeError:
+            continue
```

Ruff (0.16.2)

[warning] `try`-`except` within a loop incurs performance overhead

(PERF203)